src/main/java/io/cryostat/agent/remote/InvokeContext.java

```diff
@@ -72,6 +82,14 @@ public void handle(HttpExchange exchange) throws IOException {
                                         req.parameters,
                                         req.signature);
 
+                        // TODO: Verify if dumpHeap is blocking, if so we should split the
+                        // invocation
+                        // into a separate thread and listen for when it finishes
+                        if (req.getOperation().equals("dumpHeap")) {
+                            String fileName = req.getParameters()[0].toString();
```

Comment:
I think it would be best if the Agent automatically determines the filename that it will dump to, rather than having the server tell it what that path should be. Doing it this way seems like an unnecessary security hole where a request can be sent to the Agent to make it write the heapdump somewhere it should not be on the filesystem (ex. world readable, or a network share, etc.), or else make it write the heapdump overtop of some other important data.

I would suggest that there should be an Agent config property to set a default location for these heap dumps to be written to, and a default file name base/prefix. Appending a UUID to the base name (as is currently done on the server side) seems like a good enough way to mitigate file name conflicts.